cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -244,3 +275,108 @@ func createDNSService(dnsService *v1.Service, serviceBytes []byte, client client
 	}
 	return nil
 }
+
+// translateStubDomainOfKubeDNSToProxyCoreDNS translates StubDomain Data in kube-dns ConfigMap
+// in the form of Proxy for the CoreDNS Corefile.
+func translateStubDomainOfKubeDNSToProxyCoreDNS(dataField string, kubeDNSConfigMap *v1.ConfigMap) (string, error) {
+	if proxy, ok := kubeDNSConfigMap.Data[dataField]; ok {
+		stubDomainData := make(map[string][]string)
+		var proxyStanzaList string
+
+		err := json.Unmarshal([]byte(proxy), &stubDomainData)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse JSON from 'kube-dns ConfigMap: %v", err)
+		}
+		var proxyStanza []interface{}
+
+		for domain, proxyIP := range stubDomainData {
+			strings.Join(proxyIP, " ")
```

Comment:
This doesn't capture the return value, what is it for?

Reply:
Yes, that line is redundant. Shall remove.